core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -118,31 +128,23 @@ void HIP::impl_finalize() {
     lock.finalize();
   }
 
-  Impl::HIPInternal::singleton().finalize();
-
-  KOKKOS_IMPL_HIP_SAFE_CALL(
-      hipStreamDestroy(Impl::HIPInternal::singleton().m_stream));
+  // Destroy the default instance.
+  Impl::HIPInternal::default_instance = nullptr;
```

Comment:
Note that if we care we can now tell whether the user held on to a copy of the default instance (not proposed here)